classes/Api/ThankYous.php

```diff
@@ -0,0 +1,809 @@
+	 */
+	public function GetOwnerClassNamesFromIds(array $object_types_id)
+	{
+		return $this->utility->GetOwnerClassNamesFromIds($object_types_id);
+	}
+
+	/**
+	 * Returns an array of Users indexed by their IDs.
+	 *
+	 * @param int[] $user_ids
+	 * @return User[]
+	 */
+	public function GetUsers(array $user_ids)
+	{
+		return $this->thank_yous_repository->GetUsers($user_ids);
+	}
+
+	/**
+	 * Given an array of arrays with offset id = int, oclass = int, returns a distinct list of User IDs.
+	 *
+	 * @param array $owner_classes
+	 * @return int[]
+	 */
+	public function GetDistinctUserIdsFromOwnerClasses(array $owner_classes)
+	{
+		$acl = $this->acl_repository->Get(0, 0);
+
+		foreach ($owner_classes as $owner_class_member)
+		{
+			$id        = $owner_class_member['id'] ?? null;
+			$oclass_id = $owner_class_member['oclass'] ?? null;
+
+			if (!isset($id))
+			{
+				throw new InvalidArgumentException("Failed to Get Distinct User IDs From Owner Classes, one or more Owner Class Members does not have an ID");
+			}
+			if (!is_int($id) || !($id > 0))
+			{
+				throw new InvalidArgumentException("Failed to Get Distinct User IDs From Owner Classes, Owner Class Member ID " . (string) $id . " is not a natural number");
+			}
+
+			if (!isset($oclass_id))
+			{
+				throw new InvalidArgumentException("Failed to Get Distinct User IDs From Owner Classes, one or more Owner Class Members does not have an Owner Class ID");
+			}
+			if (!is_int($oclass_id))
+			{
+				throw new InvalidArgumentException("Failed to Get Distinct User IDs From Owner Classes, Owner Class ID " . (string) $oclass_id . " is not an integer");
+			}
+
+			try
+			{
+				$acl->Add(0, $oclass_id, $id);
+			} catch (InvalidSubjectException $invalid_subject_exception)
+			{
+				throw new LogicException("Unexpected Exception thrown by Acl method Add", null, $invalid_subject_exception);
+			}
+		}
+
+		$user_ids = $acl->GetIndividualsList(0);
+
+		foreach ($user_ids as $offset => $user_id)
+		{
+			$user_ids[$offset] = (int) $user_id;
+		}
+
+		return $user_ids;
+	}
+
+	/**
+	 * Determines which Extranets are visible to a specific Security Context, and returns their IDs.
+	 * In the event that all Extranets are visible, null is returned.
+	 *
+	 * @param SecurityContext $context
+	 * @return int[]|null
+	 */
+	public function GetVisibleExtranetIds(SecurityContext $context)
+	{
+		if ($context->IsPrimaryExtranet())
+		{
+			return null;
+		} else
+		{
+			return [(int) $context->GetExtranetAreaId(), (int) $this->extranet_service->GetPrimaryId()];
+		}
+	}
+
+	/**
+	 * @param ThankYou[] $thank_yous
+	 */
+	public function LoadThankYousComments(array $thank_yous)
+	{
+		$thank_yous_comments = [];
+		foreach ($thank_yous as $thank_you)
+		{
+			if (!($thank_you instanceof ThankYou))
+			{
+				throw new InvalidArgumentException("Failed to Load Thank Yous Comments, one or more Thank You given is not a Thank You");
```

Comment:
Just skip over anything that isn't a ThankYou.

Reply:
Fixed in 960a093eb22811448bc90946f4f1919f5ace85e6